frame/staking/src/lib.rs

```diff
@@ -1150,73 +1290,47 @@ decl_error! {
 		/// A nomination target was supplied that was blocked or otherwise not a validator.
 		BadTarget,
 	}
-}
-
-decl_module! {
-	pub struct Module<T: Config> for enum Call where origin: T::Origin {
-		/// Number of sessions per era.
-		const SessionsPerEra: SessionIndex = T::SessionsPerEra::get();
-
-		/// Number of eras that staked funds must remain bonded for.
-		const BondingDuration: EraIndex = T::BondingDuration::get();
-
-		/// Number of eras that slashes are deferred by, after computation.
-		///
-		/// This should be less than the bonding duration.
-		/// Set to 0 if slashes should be applied immediately, without opportunity for
-		/// intervention.
-		const SlashDeferDuration: EraIndex = T::SlashDeferDuration::get();
-
-		/// The maximum number of nominators rewarded for each validator.
-		///
-		/// For each validator only the `$MaxNominatorRewardedPerValidator` biggest stakers can claim
-		/// their reward. This used to limit the i/o cost for the nominator payout.
-		const MaxNominatorRewardedPerValidator: u32 = T::MaxNominatorRewardedPerValidator::get();
-
-		/// Maximum number of nominations per nominator.
-		const MaxNominations: u32 = T::MAX_NOMINATIONS;
-
-		type Error = Error<T>;
-
-		fn deposit_event() = default;
 
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
 		fn on_runtime_upgrade() -> Weight {
-			if StorageVersion::get() == Releases::V5_0_0 {
+			if StorageVersion::<T>::get() == Releases::V5_0_0 {
 				migrations::v6::migrate::<T>()
 			} else {
 				T::DbWeight::get().reads(1)
 			}
 		}
 
-		fn on_initialize(_now: T::BlockNumber) -> Weight {
+		fn on_initialize(_now: BlockNumberFor<T>) -> Weight {
 			// just return the weight of the on_finalize.
 			T::DbWeight::get().reads(1)
 		}
 
-		fn on_finalize() {
+		fn on_finalize(_n: BlockNumberFor<T>) {
 			// Set the start of the first era.
 			if let Some(mut active_era) = Self::active_era() {
 				if active_era.start.is_none() {
 					let now_as_millis_u64 = T::UnixTime::now().as_millis().saturated_into::<u64>();
 					active_era.start = Some(now_as_millis_u64);
 					// This write only ever happens once, we don't include it in the weight in general
-					ActiveEra::put(active_era);
+					ActiveEra::<T>::put(active_era);
 				}
 			}
 			// `on_finalize` weight is tracked in `on_initialize`
 		}
 
 		fn integrity_test() {
-			sp_io::TestExternalities::new_empty().execute_with(||
-				assert!(
-					T::SlashDeferDuration::get() < T::BondingDuration::get() || T::BondingDuration::get() == 0,
-					"As per documentation, slash defer duration ({}) should be less than bonding duration ({}).",
-					T::SlashDeferDuration::get(),
-					T::BondingDuration::get(),
-				)
-			);
+			assert!(
```

Comment:
is this change right?

Reply:
usually Get implementation doesn't make use of externalities, but they can do sometimes, using parameter types `pub storage ..` or some manual implementation, I think it is better not to change it

Reply:
What would happen if we tried to use storage and we didnt have this externalities? It would compile error?

Reply:
no it would panic at runtime saying that function is executed outside externalities

Reply:
looking deeper into the code, it says that this test does not execute in an externalities environment, so this removal is correct.

Also we would need to initiate the storage somehow if there was some storage used here.